php/class-customize-concurrency.php

```diff
@@ -157,29 +158,42 @@ public function preserve_inserted_post_name( $data, $postarr ) {
 	}
 
 	/**
-	 * Check all incoming values to see if any were changed between the last time we checked and when we published these
-	 * new values.
-	 *
-	 * NOTE TO SELF: This always runs, but probably should not when saving/updating snapshots, but probably should on publishing snapshot.
+	 * Set up conflict checking within snapshots.
+	 */
+	public function customize_snapshot_save_before( $snapshot, \WP_Customize_Manager $wp_customize ) {
+		$saved_settings = $snapshot->data();
+		$this->sanitize_conflicts( $saved_settings, $wp_customize );
+	}
+
+	/**
+	 * Set up conflict checking without snapshots.
 	 *
-	 * @param \WP_Customize_Manager $wp_customize Customize Manager used to get values to be checked.
+	 * @todo detect snapshot save and do not validate against live changes
```

Comment:
This function is getting run every time a snapshot gets saved, not just when settings are published. Thus snapshot saves are getting conflict tested against concurrent updates outside a snapshot. While that could be cool information, it would call for a different notice. For now we need to at least detect that we are doing a snapshot save a prevent this check.